modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,502 @@
+                  miubalibzR = miuow * _qzR[a] * -1.0 + miuox * _qyR[a] * -1.0 -
+                               miuoy * _qxR[a] * -1.0 + miuoz * _qwR[a];
+                }
+                _VlibR[_qp](0) = miubalibxR;
+                _VlibR[_qp](1) = miubalibyR;
+                _VlibR[_qp](2) = miubalibzR;
+                if (sumval == 0.0)
+                {
+                  _VlibR[_qp](0) = 0.0;
+                  _VlibR[_qp](1) = 0.0;
+                  _VlibR[_qp](2) = 0.0;
+                  miubalibwR = 0.0;
+                  miubalibxR = 0.0;
+                  miubalibyR = 0.0;
+                  miubalibzR = 0.0;
+                }
+
+                const ADReal dot_product =
+                    miubalibxR * vsmallbax + miubalibyR * vsmallbay + miubalibzR * vsmallbaz;
+                if (_thetabalib[l] < 0.0)
+                  _thetabalib[l] =
+                      6.28319 +
+                      _thetabalib[l]; // _thetabalib is in radians. 6.28319 radians is equivalent to
+                                      // 360 degrees. If the angle is negative, this converts it to
+                                      // its equivalent positive angle. -60deg around <111> =
+                                      // (360-60 = 300deg) around <111>.
+                const ADReal thetadtheta = thetaba - _thetabalib[l];
+                const ADReal normvbavalue = vbax * vbax + vbay * vbay + vbaz * vbaz;
+                const ADReal normvba = std::sqrt(normvbavalue);
+                const ADReal normvbalibvalue = _vbalibx[l] * _vbalibx[l] +
+                                               _vbaliby[l] * _vbaliby[l] +
+                                               _vbalibz[l] * _vbalibz[l];
+                const ADReal normvbalib = std::sqrt(normvbalibvalue);
+                const ADReal cosinevalue1 =
+                    _vbalibx[l] * vbax + _vbaliby[l] * vbay + _vbalibz[l] * vbaz;
+                const ADReal cosinevalue2 = normvbalib * normvba;
+                const ADReal cosinevalue3 = cosinevalue1 / cosinevalue2;
+                const ADReal thetadv = std::acos(cosinevalue3);
+                const ADReal switchvalue =
+                    -100.0 * ((thetadv / _alphaswitch) * (thetadv / _alphaswitch) +
+                              (thetadtheta / _betaswitch) * (thetadtheta / _betaswitch));
+                ADReal exp_switch_value = std::exp(switchvalue);
+                _S_switch[_qp] = exp_switch_value;
+                if (exp_switch_value < 0.001) // 0.001 is an arbitrary threshold below which the
+                                              // switch is assumed negligible.
+                  _S_switch[_qp] = 0.0;
+                const ADReal exponent = _sharpness * (dot_product - 1.0);
+                const ADReal amplitudes =
+                    (_sigmaBASE - _MinimaEnergy[l]) * _lgb / 1.0e9 * std::sqrt(f0saddle) / g_gamma;
+                const ADReal amplitudesL = std::fabs(_L_BASE - _MinimaL[l]);
```

Comment:
use `std::abs` not `std::fabs`

Reply:
It was having an issue with abs before here in the checks , so i switched to fabs.

I updated back to abs. Will see if it works.